modules/tensor_mechanics/src/materials/FiniteStrainHyperElasticViscoPlastic.C

```diff
@@ -0,0 +1,394 @@
+#include "FiniteStrainHyperElasticViscoPlastic.h"
+
+template<>
+InputParameters validParams<FiniteStrainHyperElasticViscoPlastic>()
+{
+  InputParameters params = validParams<FiniteStrainMaterial>();
+  params.addParam<Real>("resid_abs_tol",1e-10,"Absolute Tolerance for flow rate residual equation");
+  params.addParam<Real>("resid_rel_tol",1e-6,"Relative Tolerance for flow rate residual equation");
+  params.addParam<unsigned int>("maxiters", 50, "Maximum iteration for flow rate update");
+  params.addParam<unsigned int>("max_substep_iteration",1,"Maximum number of substep iteration");
+  params.addRequiredParam<UserObjectName>("stress_user_object","User object that computes stress using elastic RightCauchyGreenTensor");
+  params.addRequiredParam<std::vector<UserObjectName> >("flow_rate_user_objects","List of User object names that computes flow rate and derivatives");
+
+  return params;
+}
+
+FiniteStrainHyperElasticViscoPlastic::FiniteStrainHyperElasticViscoPlastic(const std::string & name,
+                                                                           InputParameters parameters) :
+    FiniteStrainMaterial(name, parameters),
+  _resid_abs_tol(getParam<Real>("resid_abs_tol")),
+  _resid_rel_tol(getParam<Real>("resid_rel_tol")),
+  _maxiters(getParam<unsigned int>("maxiters")),
+  _max_substep_iter(getParam<unsigned int>("max_substep_iteration")),
+  _stress_uo(getUserObject<HyperElasticStress>("stress_user_object")),
+  _num_flow_rate_uos(getParam<std::vector<UserObjectName> >("flow_rate_user_objects").size()),
+  _fp(declareProperty<RankTwoTensor>("fp")),
+  _fp_old(declarePropertyOld<RankTwoTensor>("fp")),
+  _q(declareProperty<std::vector<Real> >("internal_var")),
+  _q_old(declarePropertyOld<std::vector<Real> >("internal_var")),
+  _deformation_gradient_old(declarePropertyOld<RankTwoTensor>("deformation gradient"))
+{
+  if (_num_flow_rate_uos == 0)
+    mooseError("Specify atleast one flow rate user object");
+
+  _flow_rate_uo.resize(_num_flow_rate_uos);
+  _q_end_index.resize(_num_flow_rate_uos, 0);
+
+  UserObjectInterface uoi(parameters);
```

Comment:
This doesn't seem right? Are you adding this interface here so that you couple to other UserObjects? If so, let us know, we could probably fix that up for you.